torch/lib/c10d/NCCLUtils.hpp

```diff
@@ -16,11 +16,25 @@
     if (error != ncclSuccess) {                                           \
       std::string err = "NCCL error in: " + std::string(__FILE__) + ":" + \
           std::to_string(__LINE__) + ", " +                               \
-          std::string(ncclGetErrorString(error));                         \
+          std::string(ncclGetErrorString(error)) + ", NCCL version " +    \
+          getNcclVersion();                                               \
```

Comment:
You can turn this into a custom `ncclGetErrorString` that tacks on the version.

Then the error below will automatically use the same as well.

Reply:
Would it be confusing to override `ncclGetError` since it's implemented in the NCCL API? What about a new function `ncclGetErrorWithVersion` that wraps `ncclGetErrorString` and tacks on the version info?

Reply:
Yep, that sounds good.